python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -432,73 +433,94 @@ def _doMatch(self, refCat, sourceCat, wcs, refFluxField, numUsableSources,
         pyPPMb = PessimisticPatternMatcherB(ref_array[:, :3], self.log)
 
+        match_found = False
         # Start the iteration over our tolerances.
         for try_idx in range(self.config.matcherIterations):
-            if try_idx == 0 and \
-               match_tolerance.lastMatchedPattern is not None:
-                # If we are on the first, most stringent tolerance,
-                # and have already found a match, the matcher should behave
-                # like an optimistic pattern matcher. Exiting at the first
-                # match.
-                run_n_consent = 1
-            else:
-                # If we fail or this is the first match attempt, set the
-                # pattern consensus to the specified config value.
-                run_n_consent = numConsensus
-            # We double the match dist tolerance each round and add 1 to the
-            # to the number of candidate spokes to check.
-            matcher_struct = pyPPMb.match(
-                source_array=src_array,
-                n_check=self.config.numPointsForShapeAttempt + try_idx,
-                n_match=self.config.numPointsForShape,
-                n_agree=run_n_consent,
-                max_n_patterns=self.config.numBrightStars,
-                max_shift=maxShiftArcseconds,
-                max_rotation=self.config.maxRotationDeg,
-                max_dist=maxMatchDistArcSec * 2. ** try_idx,
-                min_matches=minMatchedPairs,
-                pattern_skip_array=np.array(
-                    match_tolerance.failedPatternList)
-            )
-
-            if try_idx == 0 and \
-               len(matcher_struct.match_ids) == 0 and \
-               match_tolerance.lastMatchedPattern is not None:
-                # If we found a pattern on a previous match-fit iteration and
-                # can't find an optimistic match on our first try with the
-                # tolerances as found in the previous match-fit,
-                # the match we found in the last iteration was likely bad. We
-                # append the bad match's index to the a list of
-                # patterns/matches to skip on subsequent iterations.
-                match_tolerance.failedPatternList.append(
-                    match_tolerance.lastMatchedPattern)
-                match_tolerance.lastMatchedPattern = None
-                maxShiftArcseconds = \
-                    self.config.maxOffsetPix * wcs.pixelScale().asArcseconds()
-                continue
-            elif len(matcher_struct.match_ids) > 0:
-                # Match found, save a bit a state regarding this pattern
-                # in the match tolerance class object and exit.
-                match_tolerance.maxShift = \
-                    matcher_struct.shift * afwgeom.arcseconds
-                match_tolerance.lastMatchedPattern = \
-                    matcher_struct.pattern_idx
+            for soften_idx in range(self.config.matcherIterations):
```

Comment:
Could you maybe clean up the language around here a bit please?

Specifically, both `soften_idx` and `try_idx` are "softening", they're just softening different things. Maybe clearer to use, say, `soften_pattern` and `soften_dist`, or something like that.